### PR TITLE
Simplify chart spec format and usage and improve empty data filtering

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: static/js/place/topic_menu.tsx [hunk 1]
```diff
@@ -15,18 +15,68 @@
...
+import { PageChart } from "./types";
+
+interface CategoryPropsType {
+  dcid: string;
+  selectCategory: string;
+  category: string;
+  topics: string[];
+}
+
+class Category extends React.Component<CategoryPropsType, unknown> {
```
Comment (beets) [inline static/js/place/topic_menu.tsx:27]: consider naming this MenuCategory or MenuTopic

File: server/routes/api/landing_page.py [hunk 2]
```diff
@@ -33,6 +34,8 @@
 # Define blueprint
 bp = Blueprint("api.landing_page", __name__, url_prefix='/api/landingpage')
 
+CHART_TYPES = ['parent', 'similar', 'nearby', 'child']
```
Comment (beets) [inline server/routes/api/landing_page.py:37]: maybe BAR_CHART_TYPES to be specific

**Review by beets — APPROVED** [review]
nice - much easier to read now!

**Comment by shifucun** [comment]
thanks for review!

**Review by beets — APPROVED** [review]

**Pull request merged**